EIPS/eip-5450.md

```diff
@@ -92,15 +92,17 @@ For each instruction:
 4. For each successor instruction:
    1. **Check** if the instruction is present in the code (i.e. execution must not "fall off" the code).
    2. If the successor is reached via forwards jump or sequential flow from previous instruction:
-      1. If the instruction does not have stack height bounds recorded (visited for the first time), record the instruction stack height bound as the value computed in 2.3.
-      2. Otherwise, instruction was already visited (by previously seen forward jump). Update this instruction's recorded stack height bounds so that they contain the bounds computed in 2.3, i.e. `target_stack_min = min(target_stack_min, current_stack_min)` and `target_stack_max = max(target_stack_max, current_stack_max)`, where `(target_stack_min, target_stack_max)` are successor bounds and `(current_stack_min, current_stack_max)` are bounds computed in 2.3.
+      1. If the instruction does not have stack height bounds recorded (being visited for the first time), record the instruction stack height bound as the value computed in 2.3.
+      2. Otherwise, instruction has been already visited (by previously seen forward jump). Update this instruction's recorded stack height bounds so that they contain the bounds computed in 2.3, i.e. `target_stack_min = min(target_stack_min, current_stack_min)` and `target_stack_max = max(target_stack_max, current_stack_max)`, where `(target_stack_min, target_stack_max)` are successor bounds and `(current_stack_min, current_stack_max)` are bounds computed in 2.3.
    3. If the successor is reached via backwards jump, **check** if the recorded stack height bounds equal the value computed in 2.3. Validation fails if they are not equal, i.e. we see backwards jump to a different stack height.
 
-After all instructions are visited, determine the function maximum operand stack height:
+After all instructions are visited, determine the function maximum operand stack height increase:
 
-1. Compute the maximum stack height as the maximum of all recorded stack height upper bounds.
-2. **Check** if the maximum stack height does not exceed the limit of 1023 (`0x3FF`).
-3. **Check** if the maximum stack height matches the corresponding code section's `max_stack_height` within the type section.
+1. Compute the maximum stack height `max_stack_height` as the maximum of all recorded stack height upper bounds.
+2. **Check** if the maximum stack height `max_stack_height` does not exceed the limit of 1024.
+3. Compute the maximum stack height increase `max_stack_increase` as `max_stack_height - type[current_section_index].inputs`.
+4. **Check** if the maximum stack height increase `max_stack_increase` does not exceed the limit of 1023 (`0x3FF`).
```

Comment:
This is redundant given 5.

Reply:
Yes, not this is more visible. Let's decide on the other comment and we can remove this point.

Reply:
Removed.